python/cuml/test/test_umap.py

```diff
@@ -112,28 +108,49 @@ def test_umap_trustworthiness_on_iris():
     embedding = cuUMAP(n_neighbors=10, min_dist=0.01,
                        verbose=False).fit_transform(data, convert_dtype=True)
     trust = trustworthiness(iris.data, embedding, 10)
-
-    # We are doing a spectral embedding but not a
-    # multi-component layout (which is marked experimental).
-    # As a result, our score drops by 0.006.
-    assert trust >= 0.964 - TRUST_TOLERANCE_THRESH
+    assert trust >= 0.97
 
 
 def test_umap_transform_on_iris():
 
     iris = datasets.load_iris()
+
     iris_selection = np.random.RandomState(42).choice(
         [True, False], 150, replace=True, p=[0.75, 0.25])
     data = iris.data[iris_selection]
 
-    fitter = cuUMAP(n_neighbors=10, min_dist=0.01, verbose=False)
+    def run_transform():
+        fitter = cuUMAP(n_neighbors=10, n_epochs=800, min_dist=0.01,
+                        verbose=False)
+        fitter.fit(data, convert_dtype=True)
+        new_data = iris.data[~iris_selection]
+        embedding = fitter.transform(new_data, convert_dtype=True)
+        return trustworthiness(new_data, embedding, 10)
+
+    threshold = 0.85
+
+    trust = run_transform()
+    if trust < threshold:
+        trust = run_transform()
+        assert trust >= threshold
```

Comment:
Do we want to only run it once? Or would a small number of runs averaged be helpful?

Reply:
Sure, i'll run it 5 times and average. After Victor's reproducibility PR, we shouldn't have to do it at all.